e2e/tests/008.sh

```diff
@@ -28,8 +29,47 @@ source "${LIBDIR}/k8s.sh"
 
 kubeconfig="$HOME/.kube/config"
 
+function k8s_upf_scale_test {
+    local current_cpu=$1
+    local current_memory=$2
+    local after_scale_cpu=$3
+    local after_scale_memory=$4
+    
+    echo "Comparing the new CPU/Memory before and after scaling"
+
+    if [ "$after_scaling_cpu" -le  "$current_cpu" ] && [ "$after_scale_memory" -le  "$current_memory" ]; then
+        echo "UPF POD Scaling Failed"
+        exit 1
+    else
+        echo "UPF Pod Scaling Successful"
+    fi
+    
+    exit 0
+}
+
+
 for cluster in "edge01" "edge02"; do
 
+    #Get the cluster kubeconfig
+    cluster_kubeconfig=$(k8s_get_capi_kubeconfig "$kubeconfig" "default" "$cluster")
+    
+    #Before scaling test get the running UPF POD ID
+    upf_pod_id=$(kubectl --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf | grep upf | cut -d ' ' -f 1)
+
+    if [ -z "$upf_pod_id" ]; then
+    	echo "UPF PoD Not Found"
+    	exit 1
+    fi
+
+    #If the pod exists, Get the current CPU and Memory limit
+    current_cpu=$(kubectl --kubeconfig $cluster_kubeconfig describe pods $upf_pod_id -n free5gc-upf | awk '/cpu:/ {print $2}' | sed 's/m$//') 
```

Comment:
First, the [resource limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) seem to be applied at container level and second, the label selection can result in multiple results.

So, in order to get the current CPU limit of all the containers for a set of pods with the same labels is:

```bash
kubectl get --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf -o jsonpath='{range .items[*].spec.containers[*]}{.resources.limits.cpu}{"\n"}{end}'
```

But if you need to get the current CPU usage, the command is:

> Note: This requires Metrics API service installed on the server.

```bash
kubectl top pod --kubeconfig $cluster_kubeconfig get pods -l app=free5gc-upf -l nf=upf -n free5gc-upf
```

Reply:
Sure, will change it to filter based on  Json path